src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -471,27 +495,58 @@ class Simplex_tree {
     auto root_source = complex_source.root_;
 
     // root members copy
-    root_.members() = Dictionary(boost::container::ordered_unique_range, root_source.members().begin(), root_source.members().end());
+    root_.members() =
+        Dictionary(boost::container::ordered_unique_range, root_source.members().begin(), root_source.members().end());
     // Needs to reassign children
     for (auto& map_el : root_.members()) {
       map_el.second.assign_children(&root_);
     }
-    rec_copy(&root_, &root_source);
+    rec_copy<Options::store_key>(
+        &root_, &root_source, [](const Filtration_value& fil) -> const Filtration_value& { return fil; });
+  }
+
+  // Copy from complex_source to "this"
+  template<typename OtherSimplexTreeOptions, typename F>
+  void copy_from(const Simplex_tree<OtherSimplexTreeOptions>& complex_source, F&& translate_filtration_value) {
+    null_vertex_ = complex_source.null_vertex_;
+    filtration_vect_.clear();
+    dimension_ = complex_source.dimension_;
+    auto root_source = complex_source.root_;
+
+    // root members copy
+    for (auto& p : root_source.members()){
+      auto it = root_.members().try_emplace(root_.members().end(), p.first);
+      it->second.assign_children(&root_);
+      it->second.assign_filtration(translate_filtration_value(p.second.filtration()));
```

Comment:
For multi, do we have that `emplace` creates a vector of size 1 containing +inf, then `translate` creates a vector of size 2 (for instance), then `assign_filtration` copies that vector, for a total of 3 allocations?

Reply:
Emplace will create an empty vector I think, but yes, that is an unnecessary number of allocations. It is just that it was how the loop was done initially. Why not do `auto it = root_.members().try_emplace(root_.members().end(), p.first, Node(&root));` instead of `auto it = root_.members().try_emplace(root_.members().end(), p.first); it->second.assign_children(&root_);` ? I was wondering if I missed something and just continued the pattern...

Reply:
Deleted old answer, as I just realized I remembered it all wrong... And I don't remember why I made this choice. I will try the constructor version, that should avoid at least one allocation.